shopify_integration/hooks.py

```diff
@@ -108,10 +108,10 @@
 
 # Overriding Methods
 # ------------------------------
-#
-# override_whitelisted_methods = {
-# 	"frappe.desk.doctype.event.event.get_events": "shopify_integration.event.get_events"
-# }
+
+override_whitelisted_methods = {
+	"frappe.integrations.doctype.connected_app.connected_app.callback": "shopify_integration.oauth.callback"
```

Comment:
Won't this break other connected apps?

Reply:
Oh, that's true, hmm. I think we'll have to look at directly overriding the redirect URI in the Connected App then. Maybe we just make it editable in the document?

Reply:
Okay, I added a validate hook that checks if the provider for the Connected App has Shopify somewhere in it's name, and then set the custom redirect URI.

Reply:
Is there no other way around this?

Reply:
We basically want a custom redirect URI for Shopify connected apps. The problem is that there's no easy way to identify whether a CA is for Shopify or otherwise.

I could add a checkbox in Connected App that says "Is Shopify App", and use that to set the redirect URI only for those apps.